components/camel-jetty9/src/main/java/org/apache/camel/component/jetty9/JettyContentExchange9.java

```diff
@@ -214,14 +220,28 @@ public void onFailure(Request request, Throwable failure) {
             }
 
         };
-        BufferingResponseListener responseListener = new BufferingResponseListener() {
+
+        InputStreamResponseListener responseListener = new InputStreamResponseListener() {
+            ByteArrayOutputStream baos = new ByteArrayOutputStream();
+
+            @Override
+            public void onContent(Response response, ByteBuffer content, Callback callback) {
+                if (response.getStatus() != SC_OK) {
```

Comment:
Ah okay but a byte array stream is unfortunately hard limited, so if onContent is called multiple times the buffer looses data etc. What is needed is to be able to store the data in a growing buffer.

You can try to do like the MarshalProcessor which uses Camel's stream cache to write to its buffer. Then when you are done, you can get the result from its build method.